blocks/api/post.pegjs

```diff
@@ -240,27 +242,81 @@ Block_Balanced
     return {
       blockName: s.blockName,
       attrs: s.attrs,
+      version: s.version,
       innerBlocks: innerBlocks,
       innerHTML: innerHTML.join( '' )
     };
   }
 
 Block_Start
-  = "<!--" WS+ "wp:" blockName:Block_Name WS+ attrs:(a:Block_Attributes WS+ {
-    /** <?php return $a; ?> **/
-    return a;
-  })? "-->"
+  = "<!--" WS+ header:Block_Header WS+ "-->"
```

Comment:
This feels a bit obscure and I'm not sure what value it brings us to pull out the `Block_Header` here into its own rule. We're not saving much repetition, and we end up with more code even I think.

Is there a reason why we created this wrapper? The wrapper rule isn't even doing anything but opaquely passing everything through `Block_Header` and so it seems to make it harder to figure out what the rule is actually doing. I would hope that as much as is possible we could keep the structure of the rules close to the structure of the document being parsed; in this case maybe that means including some repetitive bits because at a glance we can see how it's supposed to be.

Reply:
This is a factorization because the same thing is used in Block_Start and Block_Void

Reply:
Oh yeah it’s obvious what it _is_; but it’s not clear how this is any better. I was simply saying that to me it appears to obscure the grammar and I don’t find that helpful.

_Why_ is the factoring being done and how does it help someone reading or using it? If all it’s doing is removing one duplication of a few lines and it’s cost is a level of indirection that’s doing nothing then I don’t think that’s a very good bargain

Reply:
> Why is the factoring being done and how does it help someone reading or using it? I

I do think it helps people reading and maintaing the grammar. In fact when I implemented the versioning, it was broken for void blocks at first. It's easy to miss the void blocks if you're not aware they exist.

Reply:
I'd be interested in hearing some additional thoughts on that. To me it makes the grammar more confusing though I can understand where a view might be different than mine.